subprojects/gradle/runner/client/src/main/kotlin/com/avito/runner/scheduler/metrics/TestMetricsComputer.kt

```diff
@@ -0,0 +1,69 @@
+package com.avito.runner.scheduler.metrics
+
+import com.avito.logger.LoggerFactory
+import com.avito.logger.create
+
+internal class TestMetricsComputer(
+    loggerFactory: LoggerFactory,
+    private val testSuiteStartedTime: Long,
+    private val testSuiteEndedTime: Long,
+    private val onDeviceMarks: Map<TestKey, Long>,
+    startedMarks: Map<TestKey, Long>,
+    finishedMarks: Map<TestKey, Long>
+) {
+
+    private val logger = loggerFactory.create<TestMetricsComputer>()
+
+    private val firstTestStarted = startedMarks.values.minOrNull()
+
+    private val lastTestEnded = finishedMarks.values.maxOrNull()
+
+    private val queueTimes: List<Long> = onDeviceMarks.map { it.value - testSuiteStartedTime }
+
+    private val startTimes: List<Long> = startedMarks
+        .map { (key, value) ->
+            val onDeviceMark = onDeviceMarks[key]
+            if (onDeviceMark == null) {
+                logger.warn("No onDevice event for $key")
+                null
+            } else {
+                value - onDeviceMark
+            }
+        }
+        .filterNotNull()
+
+    /**
+     * null is possible if no tests
+     */
+    fun computeInitialDelay(): Long? = firstTestStarted?.let { it - testSuiteStartedTime }
```

Comment:
I think we can delete `compute` from the interface.
These are not heavy computations and clients don't need to know it.
Also, the client knows about this from the class name.

```kotlin
computer.computeInitialDelay()

computer.initialDelay()
```